###########################################################
Date: 21 May, 2020 12:55 PM
From: None
Subject: 1. SIGSEGV during NULL fingerprint comparison
License: "BSD & MIT"
Upstream-Status: Pending
Signed-off-by: Uma Kumar <Uma_Kumar@cable.comcast.com>
Patch Recreated by: Arjun <arjun_daasuramdass@comcast.com>
Patch updated to support net-snmp-5.9.1
##########################################################
Index: net-snmp-5.8/snmplib/cert_util.c
===================================================================
--- net-snmp-5.8.orig/snmplib/cert_util.c
+++ net-snmp-5.8/snmplib/cert_util.c
@@ -2548,6 +2548,9 @@ _map_compare(netsnmp_cert_map *lhs, nets
     else if (lhs->priority > rhs->priority)
         return 1;

+    /* DELIA-40163 - SIGSEGV due to string comparison with NULL fingerprint */
+    if ((lhs->fingerprint == NULL) || (rhs->fingerprint == NULL))
+        return -1;
     return strcmp(lhs->fingerprint, rhs->fingerprint);
 }

@@ -2557,6 +2560,9 @@ _map_fp_compare(netsnmp_cert_map *lhs, n
     int rc;
     netsnmp_assert((lhs != NULL) && (rhs != NULL));

+    /* DELIA-40163 - SIGSEGV due to string comparison with NULL fingerprint */
+    if ((lhs->fingerprint == NULL) || (rhs->fingerprint == NULL))
+        return -1;
     if ((rc = strcmp(lhs->fingerprint, rhs->fingerprint)) != 0)
         return rc;

@@ -2573,6 +2579,9 @@ _map_fp_ncompare(netsnmp_cert_map *lhs,
 {
     netsnmp_assert((lhs != NULL) && (rhs != NULL));

+    /* DELIA-40163 - SIGSEGV due to string comparison with NULL fingerprint */
+    if ((lhs->fingerprint == NULL) || (rhs->fingerprint == NULL))
+        return -1;
     return strncmp(lhs->fingerprint, rhs->fingerprint,
                    strlen(rhs->fingerprint));
 }

